pkg/tuf/options.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tuf
+
+import (
+	"embed"
+	"os"
+	"path/filepath"
+)
+
+//go:embed repository
+var embeddedRepo embed.FS
+
+const DefaultMirror = "https://tuf-repo-cdn.sigstore.dev"
+
+// Options represent the various options for a Sigstore TUF Client
+//
+// Note that currently, the cache control is not working. Upon initialization
+// the client will *ALWAYS* perform a TUF update.
+type Options struct {
+	// CacheValidity period in days (default 1)
+	CacheValidity int
+	// ForceCache controls if the cache should be used without update
+	// as long as the metadata is valid
+	ForceCache bool
+	// Root is the TUF trust anchor
+	Root []byte
```

Comment:
What would be the benefit of having it as a path, so it can't be modified during runtime? Or to simplify the UX for a client so they don't need to read it from disk themselves?

Reply:
This was in the context of avoiding options erroring out on initialization, to have the client handling reading the default root.

Reply:
I updated to that `DefaultOptions` is not returning an error, resolving.